test/components/admin/editions/document_history_tab_component_test.rb

```diff
@@ -16,6 +16,9 @@ class Admin::Editions::DocumentHistoryTabComponentTest < ViewComponent::TestCase
                    </nav>".html_safe
 
     Admin::Editions::DocumentHistoryTabComponent.any_instance.stubs(:paginate).returns(pagination)
+
+    params = ActionController::Parameters.new(controller: "admin/editions", action: "show", id: @document.latest_edition.id)
+    Admin::Editions::DocumentHistoryTabComponent.any_instance.stubs(:params).returns(params)
```

Comment:
having to stub this feels like a bit of a smell to me. Should probs be passed in as an arg. Although i think it could probs be picked up in a follow up PR. Wdyt?

Reply:
@davidgisbey Yeah, I agree this feels like a bit of a smell.

What about if you used `@document_history.filter` as the 'source of truth' for the currently selected filter option? That feels like it could be a nicer way to handle it, because then the view doesn't need to 'know' how to retrieve that value from the controller request params directly.

Reply:
So this is actually blowing up due to the helper method that constructs the remote pagination url using the params, rather than params being used to determine state anywhere inside the view component.

Reply:
I managed to resolve this by using the built-in route helper method `document_history_admin_edition_path` rather than the custom method `paginated_document_history_url`.

This removes our dependency on having access to the controller `params`, since we already have everything we need to generate a link with the correct URL params:

- `page` comes from `current_page`
- `editing` is a local variable
- `only` is available from `@document_history.only`